std/typecons.d

```diff
@@ -7143,3 +7143,265 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+/**
+ * Get the value of a field or property member. The member's name
+ * is resolved at runtime, rather than at compile-time.
+ *
+ * Note that currently only `pure @safe` property methods are supported.
```

Comment:
If we go down that road, it should really be four - one each for `pure`, `@safe`, `@nogc`, and `nothrow`. That seems like a lot of added complexity for not much practical gain, though.